### PR TITLE
feat: add netdata service

### LINKED ISSUE #769: netdata service
Support for running netdata service on macOS would be great

NixOS supports it: https://nixos.wiki/wiki/Netdata - and configuration is as simple as:

```nix
services.netdata = {
  enable = true;
  package = pkgs.netdataCloud;
};
```

https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/monitoring/netdata.nix

### PR DESCRIPTION
**Pull request by rsrohitsingh682**
resolves #769

### CODE REVIEW

**Comment by srid** [comment]
Would require https://github.com/NixOS/nixpkgs/pull/341472 I suppose?

**Review by srid — APPROVED** [review]
I tested this to work on a Mac. 

But note that cloud support is disabled in nixpkgs:

image

https://github.com/khaneliman/nixpkgs/blob/418918ebb1024b8cb401d7386fcc99fc8d77ac1e/pkgs/top-level/all-packages.nix#L9671

Someone should open an issue for this; and ideally fix it.

**Comment by rsrohitsingh682** [comment]
> I tested this to work on a Mac.
> 
> But note that cloud support is disabled in nixpkgs:
> 
> image
> https://github.com/khaneliman/nixpkgs/blob/418918ebb1024b8cb401d7386fcc99fc8d77ac1e/pkgs/top-level/all-packages.nix#L9671
> 
> Someone should open an issue for this; and ideally fix it.

https://github.com/NixOS/nixpkgs/pull/342708

**Comment by srid** [comment]
> NixOS/nixpkgs#342708

Tested it to work on aarch64-darwin.

**Comment by rsrohitsingh682** [comment]
Requesting review from [nix-darwin](https://github.com/LnL7/nix-darwin) maintainers on this PR.

@Enzime @emilazy @antoineco

**Comment by antoineco** [comment]
(just to clarify, I'm not a project maintainer!)

**Comment by rsrohitsingh682** [comment]
> (just to clarify, I'm not a project maintainer!

Apologies for misidentification. Thank you for correcting me.

**Comment by rsrohitsingh682** [comment]
Requesting review from ([nix-darwin](https://github.com/LnL7/nix-darwin)) maintainers again.

@emilazy @Enzime

File: modules/services/monitoring/netdata.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.netdata;
+
+in {
+  meta.maintainers = [ "rsrohitsingh682" ];
```
Comment (Enzime) [inline modules/services/monitoring/netdata.nix:7]:
```suggestion
  meta.maintainers = [ lib.maintainers.rsrohitsingh682 or "rsrohitsingh682" ];
```

File: modules/services/monitoring/netdata.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  options = {
+    services.netdata = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable Netdata Daemon.";
+      };
```
Comment (Enzime) [inline modules/services/monitoring/netdata.nix:15]:
```suggestion
      enable = mkEnableOption "Netdata daemon";
```

File: modules/services/monitoring/netdata.nix [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+      package = mkOption {
+        type = types.path;
+        default = pkgs.netdata;
+        description = "The package to use for Netdata";
+      };
```
Comment (Enzime) [inline modules/services/monitoring/netdata.nix:21]:
```suggestion
      package = lib.mkPackageOption pkgs "netdata" {};
```

File: modules/services/monitoring/netdata.nix [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    launchd.daemons.netdata = {
+      serviceConfig = {
+        Label = "netdata";
+        ProgramArguments = [
+          "/bin/sh" "-c"
+          "/bin/wait4path ${cfg.package}/bin/netdata &amp;&amp; exec ${cfg.package}/bin/netdata"
+        ];
```
Comment (Enzime) [inline modules/services/monitoring/netdata.nix:52]:
```suggestion
        command = lib.getExe cfg.package;
```

File: modules/services/monitoring/netdata.nix [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        StandardOutPath = "${cfg.logDir}/netdata.log";
+      };
+    };
+
+    environment.etc."netdata/netdata.conf".text = cfg.config;
+
+    system.activationScripts.preActivation.text = ''
+      mkdir -p ${cfg.workDir}
+    '';
+
```
Comment (Enzime) [inline modules/services/monitoring/netdata.nix:65]:
```suggestion
```

File: modules/services/monitoring/netdata.nix [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  };
+}
```
Comment (Enzime) [inline modules/services/monitoring/netdata.nix:67]:
```suggestion
}

```

Can you add a trailing newline at the end of the file?

**Review by Enzime — CHANGES_REQUESTED** [review]

**Comment by rsrohitsingh682** [comment]
@Enzime I have resolved the comments. Please let me know if anymore changes are required.

**Review by Enzime — CHANGES_REQUESTED** [review]
Can you add a test for these changes and squash the changes after that?

**Comment by rsrohitsingh682** [comment]
> Can you add a test for these changes and squash the changes after that?

@Enzime This is done.

**Review by Enzime — APPROVED** [review]
LGTM

Thanks for the contribution

**Pull request merged**